### PR TITLE
RQ-178: import, save, export and preview network logs and har files

### PR DESCRIPTION
**Pull request by nsrCodes**
superseeds PRs : #575 and #573 
so closing those for now. 

Supporting desktop app PR -> https://github.com/requestly/requestly-desktop-app/pull/28

### CODE REVIEW

File: app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/harLogs/converter.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+/**
+ * The optional chaining is so that this function is able
+ * to convert all sources of network har objects
+ * - Live from RQ proxy
+ * - importing har that was exported from RQ
+ * - importing any random (but valid) har file too!
+ */
+export const convertHarJsonToRQLogs = (har: Har): Log[] => {
+  const res: Log[] = har.log.entries.map((entry) => {
```
[the remark this answers is not in the document]
Comment (nafees87n) [reply app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/harLogs/converter.ts:84]: Yes, invalid HAR wont be imported

File: app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/harLogs/converter.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    queryString: [],
+    url: log.url,
+    headers: createHarHeaders(log.request.headers),
+  };
+  const harResponse: HarResponse = {
+    status: log.response.statusCode,
+    httpVersion: "HTTP/1.1",
+    cookies: [],
+    content: {
+      size: 33,
```
[the remark this answers is not in the document]
Comment (nsrCodes) [reply app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/harLogs/converter.ts:32]: This is dummy number for now, we do not properly calculate size, added this just to meet the spec.

[the remark this answers is not in the document]
Comment (lazyvab) [reply app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/harLogs/converter.ts:32]: Can we set it to 0 or -1? In future when we start populating the size, we could identify which ones are dummy values.

File: app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/HarImportModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  trackHarImportCompleted,
+} from "modules/analytics/events/features/sessionRecording/networkSessions";
+import { RQButton } from "lib/design-system/components";
+
+interface Props {
+  onSaved: (sessionId: string) => void;
+  btnText?: string;
+}
+
+const ImportandSaveNetworkHarModalButton: React.FC<Props> = ({ onSaved, btnText }) => {
```
Comment (lazyvab) [inline app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficExporter/HarImportModal/index.tsx:21]: Too long name. Also, not consistent with the package name, as it is a default export. Can we rename both package and the component to `HarImportModalButton`? Also, the import references can be updated.

**Review by lazyvab — APPROVED** [review]

**Pull request merged**